File: passwordvault/src/main/java/co/blustor/pwv/activities/LockingActivity.java
package co.blustor.pwv.activities;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import co.blustor.pwv.constants.Intents;
import co.blustor.pwv.database.Vault;

@SuppressLint("Registered")
public class LockingActivity extends AppCompatActivity {
    private final BroadcastReceiver lockBroadcastReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, @NonNull Intent intent) {
            String action = intent.getAction();
            if (action.equals(Intents.LOCK_DATABASE)) {
                finish();
            }
        }
    };

    @Override
    protected void onResume() {
        IntentFilter intentFilter = new IntentFilter();
        intentFilter.addAction(Intents.LOCK_DATABASE);
        registerReceiver(lockBroadcastReceiver, intentFilter);

        Vault vault = Vault.getInstance();
        if (!vault.isUnlocked()) {
            finish();
        }

        super.onResume();
    }

    @Override
    protected void onPause() {
        unregisterReceiver(lockBroadcastReceiver);
        super.onPause();
    }
}

File: gatekeeper/src/main/java/co/blustor/identity/activities/LockingActivity.kt
package co.blustor.identity.activities

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.support.v7.app.AppCompatActivity

import co.blustor.identity.constants.Intents
import co.blustor.identity.vault.Vault

@SuppressLint("Registered")
open class LockingActivity : AppCompatActivity() {

    private val lockBroadcastReceiver = object : BroadcastReceiver() {

        override fun onReceive(context: Context, intent: Intent) {
            val action = intent.action
            if (action != null) {
                if (action == Intents.lockDatabase) {
                    finish()
                }
            }
        }
    }

    override fun onResume() {
        val intentFilter = IntentFilter()
        intentFilter.addAction(Intents.lockDatabase)
        registerReceiver(lockBroadcastReceiver, intentFilter)

        if (!Vault.instance.isUnlocked) {
            finish()
        }

        super.onResume()
    }

    override fun onPause() {
        unregisterReceiver(lockBroadcastReceiver)
        super.onPause()
    }
}
